src/tests/Common/CoreCLRTestLibrary/Utilities.cs

```diff
@@ -488,5 +488,35 @@ public static void ExecuteAndUnload(string assemblyPath, string typeName, string
 
             Assert.False(alcWeakRef.IsAlive);
         }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct RLimit
+        {
+            public ulong rlim_cur; // Soft limit
+            public ulong rlim_max; // Hard limit
+        }
+
+        public const int RLIMIT_CORE = 4; // Core file size
+
+        [DllImport("libc", SetLastError = true)]
+        public static extern int setrlimit(int resource, ref RLimit rlim);
+
+        // Ensure that the OS doesn't generate core dump for the current process
+        public static void DisableOSCoreDump()
+        {
+            if ((Environment.OSVersion.Platform == PlatformID.Unix) || (Environment.OSVersion.Platform == PlatformID.MacOSX))
```

Comment:
`PlatformID.MacOSX` is deprecated - it is never returned by the API.

It may be better to use `OperatingSystem.IsLinux`, etc.

Reply:
Makes sense, I'll change it that way.